teamcity/resource_build_config.go

```diff
@@ -630,6 +675,62 @@ func flattenBuildStepCmdLine(s *api.StepCommandLine) map[string]interface{} {
 	return m
 }
 
+func flattenBuildFeatures(bfs []api.BuildFeature) ([]map[string]interface{}, error) {
+	var bfsToSave []map[string]interface{}
+	for _, bf := range bfs {
+		bfToSave := make(map[string]interface{})
+		cbf := bf.(*api.CommonBuildFeature)
+		bfToSave["type"] = cbf.Type()
+
+		if cbf.Properties() != nil && cbf.Properties().Count > 0 {
+			bfToSave["properties"] = make([]map[string]string, 0)
+			propertyList := bfToSave["properties"].([]map[string]string)
+			for _, property := range cbf.Properties().Items {
+				propertyMap := make(map[string]string)
+				propertyMap["name"] = property.Name
+				propertyMap["value"] = property.Value
+				propertyList = append(propertyList, propertyMap)
+			}
+		}
+
+		bfsToSave = append(bfsToSave, bfToSave)
+	}
+	return bfsToSave, nil
+}
+
+func expandBuildFeatures(list interface{}) ([]api.BuildFeature, error) {
+	out := make([]api.BuildFeature, 0)
+	in := list.([]interface{})
+	for _, i := range in {
+		s, err := expandBuildFeature(i)
+		if err != nil {
+			return nil, err
+		}
+		out = append(out, s)
+	}
+
+	return out, nil
+}
+
+func expandBuildFeature(raw interface{}) (api.BuildFeature, error) {
+	feature := raw.(map[string]interface{})
+	var featureType string
+	var properties []interface{}
+	if v, ok := feature["type"]; ok {
+		featureType = v.(string)
+	} else {
+		return nil, errors.New("feature type is missing")
```

Comment:
Given that `type` is required by the schema, is this possible? It might better to validate for things like empty string here instead.